### ISSUE TITLE
508-defect-2 [FOCUS MANAGEMENT, SCREENREADER]: Focus on page load SHOULD be consistent

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9305]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On the final confirmation screen, the focus is on the h1, but had been on the h2 throughout the flow. On the confirmation screen the focus **should** be on the h2 "You've successfully submitted your application." This is of consequence for the screen reader experience.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to have a consistent experience from page to page of an application, from the start to the confirmation.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Possible Fixes (optional)

Repeat the same focus management pattern as on preceding pages.

## WCAG or Vendor Guidance (optional)

* [W3C WCAG Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html)

**Comment by ala-yna** [comment]
@jenstrickland - Change should be integrated. Can you validate and close?

**Comment by jenstrickland** [comment]
Validate, focus is in the correct place. However, the yellow focus outline should not be shown for text items with focus. 

I just [messaged @1Copenut in the #vetsgov-accessibility channel](https://dsva.slack.com/archives/C8E985R32/p1592239534189400) to find out what the method others are using for this is.

**Comment by jenstrickland** [comment]
Trevor's response:
> generally teams have been adding it to their specific need with
> .element { outline: 0; }

**Comment by LinnJS** [comment]
@jenstrickland Just to confirm when we focus on elements on page load, we want to always focus on the `h2` element, and we never want a yellow highlight on it?

**Comment by jenstrickland** [comment]
@LinnJS For your application flow, yes, the h2 makes sense. We also don't generally put the yellow highlight on text in this use case. 

Double-checking with @1Copenut because I'm now doubting myself, and thinking the standard is to put it on the h1, but the h2 makes sense in this case.

**Comment by 1Copenut** [comment]
In this case, the most important thing is the H2 message telling users they have successfully submitted their form. I concur that setting focus on the H2 is the right call for this scenario.

**Comment by LinnJS** [comment]
No problemo just double checking. All the focuses are now on the `h2` the focus highlight was the part I was unsure about. But it does make sense to not have it since the `h2` is not interactive. I just have seen it so many places I was kinda used to it.

**Comment by jenstrickland** [comment]
I don't blame you for being unsure, and very much appreciate you asking. Thank you!

We put the gold halo around some headings under some circumstances. 😜  [silly phrasing intended] I think it can be subjective. We may put a focus halo on a non-interactive element when the focus is set using JavaScript and tabindex of -1, in some cases.

**Comment by LinnJS** [comment]
@jenstrickland this one is ready and on staging as well. Updated to remove gold outline, and always focusing on the h2 elements.

**Comment by jenstrickland** [comment]
Validated! 🎉
